lldb/tools/driver/CMakeLists.txt

```diff
@@ -11,6 +11,11 @@ if(APPLE)
   set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,-sectcreate,__TEXT,__info_plist,${CMAKE_CURRENT_BINARY_DIR}/lldb-Info.plist")
 endif()
 
+if (UNIX AND ${CMAKE_SYSTEM_NAME} MATCHES "AIX")                               
+  remove_definitions("-D_XOPEN_SOURCE=700")                                    
```

Comment:
This is fairly fragile. It may be better to do a `-U_XOPEN_SOURCE`.

Some comment about why you're undoing the changes [here](https://github.com/llvm/llvm-project/blob/d7ddc976d544528fe7f16882f5bec66c3b2a7884/llvm/CMakeLists.txt#L1173) might be in order as well.

Reply:
> It may be better to do a -U_XOPEN_SOURCE

.. which, at least according to my experiments, isn't possible it seems. However, also according to my experiments, `remove_definitions` doesn't actually remove macros defined by `add_compile_definitions` (only `add_definitions`), which brings up another question: Is this actually needed? Would it be sufficient to define _ALL_SOURCE as well?

Reply:
Okay, It was added by another team in IBM but I think the intention to add this way to only utilise the relevant posix standard definitions and avoid any extra un-needed definitions. But then again, some definitions being used in LLDB code, do not come under the `_X_OPEN_SOURCE` for AIX, in which case we need to use `_ALL_SOURCE` instead. But yes, I can check if only having add_definitions does not cause any compilation conflicts and update accordingly.

Reply:
Thanks. The way I'd start is the see if the `remove_definitions` call is actually doing something (in my experiments, it didn't, but I only tried it on toy programs). If it isn't, then you already have both of these defined, and there's nothing further to check.

That said, if you can get it touch with them, it might be good to check they were trying to achieve, as having different definition in different parts of the program is a recipe for problems. From that comment, it sounds like they had problems building llvm with _ALL_SOURCE, but adding the definition here still requires llvm (at least, parts included by this code) to build in this mode.